packages/devtools_app/lib/src/charts/flame_chart.dart

```diff
@@ -839,40 +832,161 @@ class FlameChartNodeWidget extends StatelessWidget {
-      ),
+  RenderObject createRenderObject(BuildContext context) {
+    return _RenderFlameRow()
+      ..nodes = nodes
+      ..zoom = zoom
+      ..startInset = startInset
+      ..chartWidth = chartWidth
+      ..selected = selected
+      ..hovered = hovered
+      ..extentDelegate = extentDelegate
+      ..controller = controller
+      ..colorScheme = Theme.of(context).colorScheme;
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, covariant _RenderFlameRow renderObject) {
+    renderObject
+      ..nodes = nodes
+      ..zoom = zoom
+      ..startInset = startInset
+      ..chartWidth = chartWidth
+      ..selected = selected
+      ..hovered = hovered
+      ..extentDelegate = extentDelegate
+      ..controller = controller
+      ..colorScheme = Theme.of(context).colorScheme;
+  }
+}
+
+class _RenderFlameRow extends RenderBox {
+   _RenderFlameRow();
+
+  _ScrollingFlameChartRowExtentDelegate? extentDelegate;
+  ScrollController? _controller;
+  set controller(ScrollController value) {
+    if (_controller == value) {
+      return;
+    }
+    _controller?.removeListener(markNeedsPaint);
+    _controller = value;
+    _controller?.addListener(markNeedsPaint);
+  }
+
+  List<FlameChartNode> get nodes => _nodes;
+  List<FlameChartNode> _nodes = [];
+  set nodes(List<FlameChartNode> values) {
+    if (identical(values, nodes)) {
+      return;
+    }
+    _nodes = values;
+    markNeedsLayout();
+  }
+
+  double get zoom => _zoom;
+  double _zoom = 0.5;
+  set zoom(double value) {
+    if (value == zoom) {
+      return;
+    }
+    _zoom = value;
+    markNeedsPaint();
+  }
+
+  double get startInset => _startInset;
+  double _startInset = 0;
+  set startInset(double value) {
+    if (value == startInset) {
+      return;
+    }
+    _startInset = value;
+    markNeedsLayout();
+  }
+
+  double get chartWidth => _chartWidth;
+  double _chartWidth = 0;
+  set chartWidth(double value) {
+    if (value == chartWidth) {
+      return;
+    }
+    _chartWidth = value;
+    markNeedsLayout();
+  }
+
+  Object? get selected => _selected;
+  Object? _selected;
+  set selected(Object? value) {
+    if (value == selected) {
+      return;
+    }
+    _selected = value;
+    markNeedsPaint();
+  }
+
+  Object? get hovered => _hovered;
+  Object? _hovered ;
+  set hovered(Object? value) {
+    if (value == _hovered) {
+      return;
+    }
+    _hovered = value;
+    markNeedsPaint();
+  }
+
+  ColorScheme get colorScheme => _colorScheme!;
+  ColorScheme? _colorScheme;
+  set colorScheme(ColorScheme value) {
+    if (value == _colorScheme) {
+      return;
+    }
+    _colorScheme = value;
+    markNeedsPaint();
+  }
+
+  @override
+  void performLayout() {
+    final width = extentDelegate!.chartWidth;
+    size = Size(
+      width,
+      constraints.maxHeight,
     );
   }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    var accumulated = 0.0;
+    var index = 0;
+    for (var node in nodes) {
```

Comment:
should we be painting only nodes in the viewport instead of all nodes in the row?

Reply:
Definitely :)